raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+# -*- coding: utf-8 -*-
+from binascii import unhexlify
+from gevent.event import AsyncResult
+from typing import List, Union, Optional
+from raiden.utils import typing
+
+import structlog
+
+from raiden.blockchain.abi import (
+    CONTRACT_MANAGER,
+    CONTRACT_TOKEN_NETWORK,
+    EVENT_CHANNEL_NEW2,
+    CHANNEL_STATE_NONEXISTENT,
+    CHANNEL_STATE_OPENED,
+    CHANNEL_STATE_CLOSED,
+    CHANNEL_STATE_SETTLED,
+)
+from raiden.constants import (
+    NETTINGCHANNEL_SETTLE_TIMEOUT_MIN,
+    NETTINGCHANNEL_SETTLE_TIMEOUT_MAX,
+)
+from raiden.network.rpc.filters import (
+    new_filter,
+    Filter,
+)
+from raiden.network.rpc.client import check_address_has_code
+from raiden.network.proxies.token import Token
+from raiden.network.rpc.transactions import (
+    check_transaction_threw,
+    estimate_and_transact,
+)
+from raiden.exceptions import (
+    DuplicatedChannelError,
+    InvalidSettleTimeout,
+    SamePeerAddress,
+    ChannelBusyError,
+    TransactionThrew,
+)
+from raiden.settings import (
+    DEFAULT_POLL_TIMEOUT,
+)
+from raiden.utils import (
+    address_decoder,
+    address_encoder,
+    encode_hex,
+    isaddress,
+    pex,
+    privatekey_to_address,
+    releasing,
+)
+
+log = structlog.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class TokenNetwork:
+    def __init__(
+            self,
+            jsonrpc_client,
+            manager_address,
+            poll_timeout=DEFAULT_POLL_TIMEOUT):
+        # pylint: disable=too-many-arguments
+
+        if not isaddress(manager_address):
+            raise ValueError('manager_address must be a valid address')
+
+        check_address_has_code(jsonrpc_client, manager_address, 'Channel Manager')
+
+        proxy = jsonrpc_client.new_contract_proxy(
+            CONTRACT_MANAGER.get_abi(CONTRACT_TOKEN_NETWORK),
+            address_encoder(manager_address),
+        )
+
+        CONTRACT_MANAGER.check_contract_version(
+            proxy.call('contract_version').decode(),
+            CONTRACT_TOKEN_NETWORK
+        )
+
+        self.address = manager_address
+        self.proxy = proxy
+        self.client = jsonrpc_client
+        self.node_address = privatekey_to_address(self.client.privkey)
+        self.poll_timeout = poll_timeout
+        self.open_channel_transactions = dict()
+
+    def _call_and_check_result(self, function_name: str, *args):
+        call_result = self.proxy.call(function_name, *args)
+
+        if call_result == b'':
+            self._check_exists()
+            raise RuntimeError(
```

Comment:
I don't have a strong opinion on this, I mainly reused the logic from the netting channel proxy. So you should not use `_call_and_check_result` if you don't expect to have a result there. If somehow you get empty bytes, then it can either be a programming error in the Raiden client/smart contracts (`assert` is better) or.. something else (probably why `RuntimeError` was used).
I'll change this to an `assert` if you think it is better.